pkg/pquerier/collector.go

```diff
@@ -204,3 +215,57 @@ func downsampleRawData(ctx *selectQueryContext, res *qryResults,
 
 	return lastT, lastV, nil
 }
+
+func aggregateClientAggregatesCrossSeries(ctx *selectQueryContext, res *qryResults, previousPartitionLastTime int64, previousPartitionLastValue float64) (int64, float64, error) {
+	ctx.logger.Debug("using Client Aggregates Collector for metric %v", res.name)
+	it := newRawChunkIterator(res, nil).(*rawChunkIterator)
```

Comment:
Pass the logger in? `newRawChunkIterator` should already return a `rawChunkIterator`.